common/define-grammar.js

```diff
@@ -496,6 +509,21 @@ module.exports = function defineGrammar(dialect) {
         $.constructor_type
       ),
 
+      tuple_rest_member: $ => $._rest_identifier,
+
+      optional_tuple_type: $ => seq($._type, '?'),
+
+      _tuple_type_member_type: $ => choice($.optional_tuple_type, $._type),
+
+      labeled_tuple_type_member: $ => seq($._tuple_type_member_type, $.type_annotation),
```

Comment:
In a labeled tuple type member, I don't think the *label* should be parsed as a type. For example, in `[a: T, b: U]`, the names `a` and `b` are not types - they should just be parsed as `identifier`.

So I think this rule should just be something like:

```js
seq(
  $.identifier,
  optional('?'),
  $.type_annotation
)
```

Reply:
Yes, having everything as type was wrong. I've changed `_tuple_type_member_type` and `optional_tuple_type` to host `$.identifier` instead of `$._type` and now it seems to be better.

Reply:
Following this suggest, I'm working on a new design for those types. Will move this PR back to Ready when it's done.